naga/src/back/hlsl/writer.rs

```diff
@@ -170,12 +170,14 @@ impl<'a, W: fmt::Write> super::Writer<'a, W> {
         }
 
         let loop_bound_name = self.namer.call("loop_bound");
-        let decl = format!("{level}uint2 {loop_bound_name} = uint2(0u, 0u);");
-        let level = level.next();
         let max = u32::MAX;
+        // Count down from u32::MAX rather than up from 0 to avoid hang on
+        // certain Intel drivers. See <https://github.com/gfx-rs/wgpu/issues/7319>.
```

Comment:
style: I've always found angle-bracketed links to be easier to work with. Doesn't really matter, tho. 

Pushed as a `fixup!` with b4f25d22.